0000-classical-stores.md

```diff
@@ -0,0 +1,493 @@
+That made sense for `Qubit` tracking, where there's no distinction between "read" and "write" operations, only uses, but a variable is typically read far more than it's written, and we do not want to add spurious data-flow restrictions between operations that all just read from the same memory without writing to it.
+
+Instead, we can insert a directed edge from a write to the variable to all the reads that it is live for.
+On the next write, including if a control-flow block may conditionally write to a closed-over variable, all the reads add an edge point from themselves to that write, the original definition becomes dead, and the new write becomes the source of wires to subsequent uses.
+Loop bodies do not need a self wire here; it's not needed for the data-flow tracking that Qiskit is concerned with.
+
+This is a crude approximation at proper SSA tracking of variables, but this should be sufficient, if a little over-cautious; certain assignments in deeply nested conditions could introduce ordering relations that could be somewhat simplified.
+In general, though, there's only so much Qiskit should attempt at this point.
+
+
+### The `Store` instruction
+
+`QuantumCircuit.store` is a shorthand for emitting a `Store` subclass of `Instruction`.
+For further discussion on why this is a subclass of `Instruction` and not a more fundamental modification to `QuantumCircuit.data` or `CircuitInstruction`, see the "alternatives" section below.
+
+The two parameters of `Store` are the location and the value.
+These are stored in `Store.params`, like a parametrised gate object, so that they can be moved to a more suitable location in a similar manner in the future.
+It is not expected that any code that currently inspects `Instruction.params` should be able to automatically work with `Store`; these objects require special handling, but this is generally true for the `.params` field which is (unfortunately) quite unconstrained in its typing.
+
+
+## Alternative Approaches
+
+### Large-scale alternatives
+
+#### Expand `condition` or `Expr` fields to support `ParameterExpression` and use `assign_parameters`
+
+As discussed in [RFC 10](0010-simple-classical-representations.md), reuse of `ParameterExpression` is not suitable for typed inputs.
+It also fundamentally represents a symbolic calculation on real numbers, rather than a circuit run-time evaluation of an object, so it would be close to impossible to align the two computational models.
+
+#### Give `Var` nodes backing `Clbit`s
+
+I seriously considered this to the degree of holding up publishing this RFC to pivot to this design.
+It has the benefit of:
+- making `ClassicalRegister` a little less special-cased; it's a variable with contained bits.
+- it would be possible to represent a `measure` directly into a bit of a new-style variable
+
+However, in the end, I have many reasons that I think this would not the correct decision:
+
+- When 
```

Comment:
Should this bullet be removed?

Reply:
Yes - I _knew_ there was somewhere I'd left an unfinished thought as I was pushing it, but I just couldn't spot it!

Reply:
Fixed in 65f9fa2d33e8405186f92e54b5cb25a53beb6e17.